lib/pbench/cli/agent/commands/generate_token.py

```diff
@@ -0,0 +1,71 @@
+"""pbench-generate-token"""
+
+import click
+import json
+import requests
+
+from pbench.agent.base import BaseCommand
+from pbench.cli.agent import pass_cli_context
+from pbench.cli.agent.options import common_options
+
+
+class GenerateToken(BaseCommand):
+    def __init__(self, context):
+        super().__init__(context)
+
+    def execute(self):
+        """Generate a token used to access the pbench server RESTful API"""
+
+        # FIXME:  Should we be logging anything here?
+        #
+        #  self.logger.warn(
+        #     "%s deprecated, will be removed in future release in favor of pbench-clear-results",
+        #     self.name,
+        # )
+
+        server = self.config.agent.parser.get("results", "server_rest_url")
+        headers = {"Content-Type": "application/json", "Authorization": None}
```

Comment:
I think I'd rather omit the `Authorization` header for login rather than supplying one but omitting the value. That just leads to questions about how that's handled in the code that make me want to go re-read Nikhil's login for no real purpose in this context. Let's try to avoid red herrings like that.

And if we're going to be "properly proper" here, maybe an `Accept: application/json` would be good; our current APIs will do that regardless, but establishing good habits is rarely a bad thing.